azerothcore/azerothcore-wotlk
# fix(DB/gameobject): fix various overlapping Water Barrels/Food Crates

## Changes Proposed:

For each of the five locations, this puts both gameobjects into a pool where only one can spawn at a time and gives them an equal spawn chance.

This is a continuation of my previous PR #6363 (where I already fixed another another such instance of overlapping Water Barrels/Food Crates), during which I discovered these other locations (see [here](https://github.com/azerothcore/azerothcore-wotlk/pull/6363#issuecomment-861900753) for details).

## Issues Addressed:

- Closes #6414

## SOURCE:

I couldn't find any specific information for these particular locations, so it may well be that some of those gameobjects shouldn't even exist there.
But it should be uncontroversial that the gameobjects should at least not overlap.

## Tests Performed:

See _How to Test the Changes_.

## How to Test the Changes:

Travel (`.go o`) to each of the following gameobjects: `29275`, `32313`, `30677`, `29276`, `29609` and complete the following steps:

1. Observe that either the Water Barrel or Food crate is spawned, but not both at the same time
2. Loot the Water Barrel or Food Crate
3. Wait a few minutes for the respawn and observe that either the Water Barrel or Food crate gets spawned, but not both at the same time

## How to Test AzerothCore PRs

When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
